sdk/python/kfp/compiler/_op_to_template.py

```diff
@@ -288,6 +289,14 @@ def _op_to_template(op: BaseOp):
         template['volumes'] = [convert_k8s_obj_to_json(volume) for volume in processed_op.volumes]
         template['volumes'].sort(key=lambda x: x['name'])
 
+    # resource requests at runtime
+    if processed_op.cpu_request and not processed_op.memory_request:
```

Comment:
Can we just build the patch piece by piece independently?
We can build a dict, then serialize it to JSON.
That can make the code easier and less fragile.

Reply:
Good point, update.